mapie/estimator/interface.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+from typing import Optional, Tuple, Union
+
+from sklearn.base import RegressorMixin
+
+from mapie._typing import ArrayLike, NDArray
+
+
+class EnsembleEstimator(RegressorMixin):
+    """
+    This class implements methods to handle the training and usage of the
+    estimator. This estimator can be unique or composed by cross validated
+    estimators.
+    """
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> EnsembleEstimator:
+        """
+        Fit the base estimator under the ``single_estimator_`` attribute.
+        Fit all cross-validated estimator clones
+        and rearrange them into a list, the ``estimators_`` attribute.
+        Out-of-fold conformity scores are stored under
+        the ``conformity_scores_`` attribute.
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+        y: ArrayLike of shape (n_samples,)
+            Input labels.
+
+        sample_weight: Optional[ArrayLike] of shape (n_samples,)
+            Sample weights. If None, then samples are equally weighted.
+            By default ``None``.
+
+        Returns
+        -------
+        EnsembleRegressor
+            The estimator fitted.
+        """
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        return_multi_pred: bool = True
+    ) -> Union[NDArray, Tuple[NDArray, NDArray, NDArray]]:
+        """
+        Predict target from X. It also computes the prediction per train sample
+        for each test sample according to ``self.method``.
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        ensemble: bool
+            Boolean determining whether the predictions are ensembled or not.
+            If ``False``, predictions are those of the model trained on the
+            whole training set.
+            If ``True``, predictions from perturbed models are aggregated by
+            the aggregation function specified in the ``agg_function``
+            attribute.
+
+            If ``cv`` is ``"prefit"`` or ``"split"``, ``ensemble`` is ignored.
+
+            By default ``False``.
+
+        return_multi_pred: bool
+
```

Comment:
UPD: add a description for this parameter.

Reply:
Done